### ISSUE TITLE
[Bug] Infinite loop in tcprewrite at get.c:569

### ISSUE THREAD

**Issue by TimChan2001** [appneta/tcpreplay#827]
**Describe the bug**
Tcprewrite gets stuck in an infinite loop at the while loop at get.c:569.

**To Reproduce**
1. export CC=clang && export CFLAGS="-fsanitize=address -g"
2. ./autogen.sh && ./configure --disable-shared --disable-local-libopts && make clean && make -j8
3. ./src/tcprewrite -o /dev/null -i POC
The POC file can be downloaded in [POC](https://github.com/TimChan2001/pocs/raw/main/poc-tcpreplay-tcprewrite-hang)

**Expected behavior**
The program continues to run for over a week without termination.
<img width="392" alt="image" src="https://github.com/appneta/tcpreplay/assets/61645944/67c78616-abe7-413f-ad29-1b95762d5585">

**System (please complete the following information):**
 - OS: Ubuntu
 - OS version: 16.04, 64 bit
 - Tcpreplay Version: 4.4.4

**Additional context**
After debugging with GDB and adding debug outputs, it seems the program is unable to exit the while loop at get.c:569.

**Comment by GabrielGanne** [comment]
Hi @TimChan2001 
I have proposed a small fix for the issue you raised.
I should have tested on the same env you described but got an different error with your pcap  first:
```
Fatal Error: Unable to open input pcap file: unsupported pcap savefile version 60418.4
```
I had to rewrite it to first and then reproduced the issue.
The fix seems straightforward enough, I expect it will work for you.
BR,

**Comment by fklassen** [comment]
Not merged yet - reopening

**Comment by fklassen** [comment]
fixed in PRs #842 and #859
